api/python/cellxgene_census/src/cellxgene_census/_open.py

```diff
@@ -123,11 +135,20 @@ def open_soma(
     """
 
     if uri is not None:
-        return _open_soma({"uri": uri, "s3_region": None}, context)
+        return _open_soma({"uri": uri, "region": None, "provider": "file"}, context)
```

Comment:
this is confusing to me - the provider is not necessarily a file (see the examples above, where several types of URIs are used -- s3:, file:, etc).

Currently `provider` does nothing - it is there to allow future code to decide how to compose a fully resolved URI and/or which API/protocol to use.  So this works, but will likely fail in the future.

I think we need to do one of two things:
* Have the code guess/determine the provider based on the value of the `uri=` param.
* Or, define an `unknown` provider, which tries its best, but may fail in some cases
* or?

Either way, if the user provides `open_soma(uri="s3://...")`, it seems pretty weird to call this a `file` provider.

Reply:
My assumption here is that `uri` was used to specify local files, but I now realize that this is more generic and can also be used to point to a file hosted in S3 (not by us). I'll go with `unknown` for now.

Reply:
Sounds good.  This further motivates the design doc request, indicating how mirrors.json and release.json fields should be combined to resolve the actual data....